src/plugins/data/common/search/tabify/tabify.ts

```diff
@@ -54,7 +54,12 @@ export function tabifyAggResponse(
       switch (agg.type.type) {
         case AggGroupNames.Buckets:
           const aggBucket = get(bucket, agg.id);
-          const tabifyBuckets = new TabifyBuckets(aggBucket, agg.params, respOpts?.timeRange);
+          const tabifyBuckets = new TabifyBuckets(
+            aggBucket,
+            agg.params,
+            respOpts?.timeRange,
+            agg.type.name === 'filter'
```

Comment:
I'd appreciate another look at this from @lukeelmers or others: I need to identify "normal" aggregations vs unusual ones, and the `filter` agg is the one that was called out in docs. I think I have two options:

1. Special case for `filter`, like this
2. Add a property to `AggType` to indicate that it's an unusual response shape

Reply:
To summarize our offline chat -- in general I would prefer to wait and see what the ES team says about https://github.com/elastic/elasticsearch/issues/63842. If it's something that can be addressed shorter-term, I'd say let's just hold off on this PR.

If it can't be addressed shorter term, I would vote for addressing this as part of a wider tabify refactor... I think right now tabify knows too much about aggs, and we could probably find a way to move some of this logic over to individual agg types. (Would need to think about it, but perhaps something like a method on each agg type that can convert its response into a standardized tabify-compatible format).

So in that sense, I think an approach closer to option 2 feels like the right direction, but perhaps something more extensive than a flag that just says whether the agg response is "normal" or not.

I'm also not sure which agg types besides filter might match this condition. Would need to look into each of them individually, unless anyone else already knows (@ppisljar ?)

Reply:
Based on the latest discussion with the ES team, we're looking at handling this in ES. It's possible that response filtering is not the right mechanism for this, but I've also opened a feature request specifically for the date_histogram to have a way to remove the key_as_string parameter.